logic.py

```diff
@@ -116,11 +122,20 @@ def get_paste_listing(config):
         return "ERROR", "Paste listing has been disabled by the " +\
             "administrator.", 503
 
+    b = config['b']
+
     try:
-        paste_list = config['b'].get_all_paste_ids()
-    except config['b'].e.ErrorException as errmsg:
+        paste_list = b.get_all_paste_ids()
+    except b.e.ErrorException as errmsg:
         return "ERROR", errmsg, 500
 
+    if (paste_list[0] != "none"):
+        # filter pastes to keep the public ones only
+        paste_list = [ paste for paste in paste_list 
```

Comment:
I was actually thinking about this. While it works for small amount of pastes and it's the same in the current backend, do you think we should implement a new method in the backend that returns all pastes where metadata key `k` has value `v`? If we add in the future a MySQL backend it will have to run a single query and not `n` queries where `n` is the number of stored pastes. What do you think?